api/api/models/media.py

```diff
@@ -183,60 +180,121 @@ class AbstractMediaReport(models.Model):
     )
     status = models.CharField(max_length=20, choices=STATUS_CHOICES, default=PENDING)
+    """
+    All statuses except ``PENDING`` are deprecated. Instead refer to the
+    properties ``is_pending`` and ``is_moderated``.
+    """
+
+    decision = models.ForeignKey(
+        to="AbstractMediaDecision",
+        on_delete=models.SET_NULL,
+        blank=True,
+        null=True,
+        help_text="The moderation decision for this report.",
+    )
 
     class Meta:
         abstract = True
 
     def clean(self):
         """Clean fields and raise errors that can be handled by Django Admin."""
 
-        if not self.media_class.objects.filter(
-            identifier=self.media_obj.identifier
-        ).exists():
+        if not self.media_class.objects.filter(identifier=self.media_obj_id).exists():
             raise ValidationError(
-                f"No '{self.media_class.__name__}' instance"
-                f"with identifier {self.media_obj.identifier}."
+                f"No '{self.media_class.__name__}' instance "
+                f"with identifier '{self.media_obj_id}'."
             )
 
-    def url(self, media_type):
-        url = f"{settings.CANONICAL_ORIGIN}v1/{media_type}/{self.media_obj.identifier}"
+    def url(self, request=None) -> str:
+        """
+        Build the URL of the media item. This uses ``reverse`` and
+        ``request.build_absolute_uri`` to build the URL without having to worry
+        about canonical URL or trailing slashes.
+
+        :param request: the current request object, to get absolute URLs
+        :return: the URL of the media item
+        """
+
+        url = reverse(
+            f"{self.media_class.__name__.lower()}-detail",
+            args=[self.media_obj_id],
+        )
+        if request is not None:
+            url = request.build_absolute_uri(url)
         return format_html(f"<a href={url}>{url}</a>")
 
-    def save(self, *args, **kwargs):
+    @property
+    def is_pending(self) -> bool:
+        """
+        Determine if the report has not been moderated and does not have an
+        associated decision. Also see ``is_moderated``.
+
+        :return: whether the report is in the "pending" state
         """
-        Save changes to the DB and sync them with Elasticsearch.
 
-        Extend the built-in ``save()`` functionality of Django with Elasticsearch
-        integration to update records and refresh indices.
+        return self.decision_id is None
 
-        Media marked as sensitive or deleted also leads to instantiation of their
-        corresponding sensitive or deleted classes.
+    @property
+    def is_moderated(self) -> bool:
         """
+        Determine if the report has been moderated and has an associated
+        decision. Also see ``is_pending``.
 
-        self.clean()
+        :return: whether the report is in the "moderated" state
+        """
 
+        return self.decision_id is not None
```

Comment:
It seems duplicative of `not self.is_pending`. It's probably clearer for code needing this information to just reference `is_pending`?

This isn't used yet, so we could just leave it out for now until we need it ([following YAGNI principles](https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it))

Reply:
I only had them because it felt "more readable" (subjective) to write report.is_pending or report.is_moderated because Python's `not` comes before the noun. 

But I concede, it was not good practice to have two ways to do one thing and also YAGNI. Dropped in 6a806b9.